### DISCUSSION TITLE
Applying track labels to predictions/instances

### DISCUSSION THREAD

**Discussion by rfkova** [talmolab/sleap#758] [Help!] [answered]
Hello!

How can I apply a track to every instance in a single animal tracking context? 'Propagate Track Labels' option does not appear to work on already labeled data unless I misunderstand it's function; do I have to apply tracks manually to already labeled data? What about predictions, they are showing no labels at present even after putting track numbers on a few labeled frames and running inference with tracking.

Cheers,
Ryan

**Comment by talmo** [comment, marked as answer]
Hi @rfkova,

Why do you need the track set for a single instance tracking model? It's implicit if there is only a single animal per frame. Is it for compatibility with downstream analyses or something?

In any case, here's a quick script that you should be able to run on Colab or in a local script or notebook to do this:

```py
import sleap

labels = sleap.load_file("labels.v000.slp")
track = sleap.Track(name="animal")  # the name is optional and be whatever you want
for instance in labels.instances():
    instance.track = track
labels.save("labels.v001.slp")
```

The propagate track labels option refers to manually switching track identities to fix swaps that happen during multi-instance tracking. It just means that switching the tracks in one frame will also be applied in all subsequent frames. Not really relevant here if I understand correctly.

Talmo

**Reply by rfkova** [reply]
This is Ryan by the way (we've been emailing). I came over here because this was a general use question and didn't want to crowd your inbox with too many basic questions. 

In any case, yes, I'm trying to think ahead to adding pup labels and realizing I'll have to have some way of integrating labels when multiple animals get involved. I was also troubleshooting SIMBA imports but it turned out I was trying to import the wrong .slp file (project and export, not predictions).

Thank you!!!
Ryan

**Reply by rfkova** [reply]
Actually, while I have you - I ran predictions 1000 at a time until I got about 50% of my videos labeled. I wanted to export these to SIMBA and interpolate intermediate frames while we are figuring out why predictions on the whole video weren't functioning.

I come to find out the prediction files are generated each time I run inference... I have two videos in this project but SIMBA wants predictions files named for each video... is there a way for me to delete one of the videos in the project and make a prediction file with all predictions therein that I could use in SIMBA? I tried deleting one video and making an export package with labels+predictions+suggestions but SIMBA didn't like that.

**Reply by talmo** [reply]
It sounds like what you want is to break out a multi-video project into individual SLP files. Definitely doable -- do you mind just copying this Q into a new thread?

**Reply by rfkova** [reply]
Absolutely!
